crates/fiber/src/lib.rs

```diff
@@ -1,13 +1,32 @@
+#![no_std]
+
+// no_std support: we can only support `panic=abort` builds, because the
+// `core::panic` module otherwise does not provide us the catch/resume
+// unwinding primitives we need to propagate unwinding across stacks.
+#[cfg(all(panic = "unwind", not(feature = "std")))]
+compile_error!(
+    "no_std builds with wasmtime-fiber (e.g., Wasmtime async) only support `panic=abort`."
+);
+
+#[cfg(any(feature = "std", unix, windows))]
+#[macro_use]
+extern crate std;
+extern crate alloc;
+
+use alloc::boxed::Box;
 use anyhow::Error;
-use std::any::Any;
-use std::cell::Cell;
-use std::io;
-use std::marker::PhantomData;
-use std::ops::Range;
+use core::any::Any;
+use core::cell::Cell;
+use core::marker::PhantomData;
+use core::ops::Range;
+#[cfg(feature = "std")]
 use std::panic::{self, AssertUnwindSafe};
```

Comment:
Mind scoping this import to just where it's needed below? (avoid the duplicate `#[cfg]`)

Reply:
Done.